server/src/main/java/org/opensearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -101,6 +101,22 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         builder.endObject();
     }
 
+    /**
+     * Adds a filter to the current ConstantScoreQuery.
+     * @param filter the filter to add to the current ConstantScoreQuery
+     * @param filterCombinationMode the combination mode for the filter. Currently support {@link FilterCombinationMode#AND}
+     */
+    public ConstantScoreQueryBuilder filter(QueryBuilder filter, FilterCombinationMode filterCombinationMode) {
+        if (!validateFilterParams(filter, filterCombinationMode)) {
+            return this;
+        }
+        QueryBuilder filteredFilterBuilder = filterBuilder.filter(filter, filterCombinationMode);
+        if (filteredFilterBuilder != filterBuilder) {
+            return new ConstantScoreQueryBuilder(filteredFilterBuilder);
+        }
+        return this;
```

Comment:
May read better testing for equality and having `return this;`  with the final return below.

However, I'm really not sure when the equality will exist.  It seems it's only in the case where `filter` is null which is already handled in line 110-112?  Or is this to catch other implementations that may not do the null-check?

Reply:
Yeah, I see @msfroh commented about that at https://github.com/opensearch-project/OpenSearch/pull/17409#discussion_r1968537361.  This is fine.